### PR TITLE
DocFx implementation

### LINKED ISSUE #160: Publish Documentation
Consider using [docfx](https://dotnet.github.io/docfx/) for documentation once we complete documentation the code.

Maybe add code samples in  the documentation like [cake](https://github.com/cake-build/cake) does.
example:
souce - https://github.com/cake-build/cake/blob/develop/src/Cake.Common/ArgumentAliases.cs#L28
generated docs - https://cakebuild.net/api/Cake.Common/ArgumentAliases/20DF3649

### PR DESCRIPTION
**Pull request by kblok**
closes #160

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -5,9 +5,16 @@ branches:
 image: Visual Studio 2017
 configuration: Release
 environment:
+  git_access_token:
+    secure: FxcQ9C8a/NgcQB5dFdZts6ZWEDT4zMhA4qPQAYwWc7huMmhmTIl1sbFEIaAWQMTL
```
Comment (Meir017) [inline appveyor.yml]: is this on purpose?

File: appveyor.yml [hunk 2]
```diff
@@ -5,9 +5,16 @@ branches:
...
 environment:
+  git_access_token:
+    secure: FxcQ9C8a/NgcQB5dFdZts6ZWEDT4zMhA4qPQAYwWc7huMmhmTIl1sbFEIaAWQMTL
   matrix:
     - framework: net471
     - framework: netcoreapp2.0
+install:
+- ps: >-
+    if($env:APPVEYOR_REPO_TAG -eq 'True' -And $env:framework -eq 'netcoreapp2.0') {
+        choco install docfx
```
Comment (Meir017) [inline appveyor.yml]: is `choco` already installed?

File: docfx_project/api/index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# Puppeteer Sharp
```
Comment (Meir017) [inline docfx_project/api/index.md]: isn't the content of this file already in `docfx_project/index.md`?

File: appveyor.yml (hunk 1 above)
Comment (kblok) [reply appveyor.yml]: Yes, it's encrypted.

File: appveyor.yml (hunk 2 above)
Comment (kblok) [reply appveyor.yml]: Yes.

File: docfx_project/api/index.md (hunk 3 above)
Comment (kblok) [reply docfx_project/api/index.md]: docfx_project/index.md => Site home => https://www.puppeteersharp.com/
docfx_project/api/index.md => It's the API home  => https://www.puppeteersharp.com/api/

Comment (Meir017) [reply docfx_project/api/index.md]: can we embed it into the other page?

**Comment by Meir017** [comment]
http://www.puppeteersharp.com/api/PuppeteerSharp.Page.html currently shows the inherited method from the `Object` class

image

**Comment by kblok** [comment]
@Meir017 site republished.

**Pull request merged**